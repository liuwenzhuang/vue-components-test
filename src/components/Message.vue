<template>
  <li class="message" style="margin-top: 10px;" @click="onMessageClick">
    {{ message }}
  </li>
</template>

<script>
export default {
  props: {
    message: String,
    author: {
      type: String,
      default: 'LWZ',
      validator: author => author.length > 1
    }
  },
  methods: {
    onMessageClick() {
      this.$emit('message-click', this.message)
    }
  }
}
</script>
