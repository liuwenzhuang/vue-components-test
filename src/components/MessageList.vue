<template>
  <ul>
    <Message
      v-for="message in messages"
      :message="message"
      :key="message"
      @message-click="onMessageClick"
    />
  </ul>
</template>

<script>
import Message from './Message.vue'

export default {
  props: ['messages'],
  components: {
    Message
  },
  methods: {
    onMessageClick(message) {
      console.log(message)
    }
  }
}
</script>
