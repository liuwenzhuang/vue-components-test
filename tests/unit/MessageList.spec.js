import { mount } from '@vue/test-utils'
import MessageList from '@/components/MessageList.vue'
import Message from '@/components/Message.vue'

describe('MessageList.vue', () => {
  let cmp
  const messages = ['New Msg1', 'New Msg2']
  beforeEach(() => {
    cmp = mount(MessageList, {
      propsData: { messages }
    })
  })
  it('should rewrite messages when passed', () => {
    expect(cmp.vm.messages).toEqual(messages)
  })
  it('should contain 2 li element', () => {
    expect(cmp.findAll('li').length).toBe(2)
    expect(
      cmp
        .findAll('li')
        .at(0)
        .text()
    ).toMatch('New Msg1')
  })
  it('should has ul tag', () => {
    expect(cmp.element.tagName).toBe('UL')
  })
  it('should contains Message Component or li elements', () => {
    expect(cmp.findComponent(Message).vm).toBeTruthy()
    expect(cmp.find('li')).toBeTruthy()
  })
  it('Message is not empty', () => {
    expect(cmp.findComponent(Message).element).toBeTruthy()
  })
  it('Message should has .message class', () => {
    expect(cmp.findComponent(Message).classes()).toContain('message')
  })
  it('Message should has style margin-top:10', () => {
    expect(cmp.findComponent(Message).attributes().style).toBe(
      'margin-top: 10px;'
    )
  })
  it('should has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})
