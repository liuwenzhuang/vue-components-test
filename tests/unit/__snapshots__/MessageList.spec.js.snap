// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.vue should has the expected html structure 1`] = `
<ul>
  <li
    class="message"
    style="margin-top: 10px;"
  >
    
  New Msg1

  </li>
  <li
    class="message"
    style="margin-top: 10px;"
  >
    
  New Msg2

  </li>
</ul>
`;
